client/src/components/RoleBasedSolution.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useEffect, useState } from 'react';
+import Solution from '../pages/Solution.tsx';
+import AdminSolution from '../pages/AdminSolution.tsx';
+
+const RoleBasedSolution = () => {
+  const [isAdmin, setIsAdmin] = useState(false);
+
+  //fetch the username from the server
+  useEffect(() => {
+    fetch("/api/users/whoami")
```

Comment:
does whoami return the role as well? I thought it only returned the utorid?

Reply:
I updated it to return the role too